### PR TITLE
Performance tests with Locust

### PR DESCRIPTION
**Pull request by cpcundill**
Added performance test for both upload and url submission across all dataset scenarios.

Implemented in Python code using the Locust framework.

### CODE REVIEW

File: performance-test/locustfile.py [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+def on_locust_init(environment: Environment, **kwargs):
+    logging.info(f"on_locust_init called - loading config file")
+    with open("config.json") as configfile:
+        global config
+        config = Config.model_validate_json(configfile.read())
+
+
+class DataProviderUser(FastHttpUser):
+
+    host = 'https://publish.development.digital-land.info'
```
Comment (GeorgeGoodall) [inline performance-test/locustfile.py:26]: is this host used anywhere?

**Review by GeorgeGoodall — APPROVED** [review]
looks good but one small comment

File: performance-test/requirements.txt [hunk 2]
```diff
@@ -0,0 +1,3 @@
+locust==2.24.1
+pydantic>=2.6.1
+har2locust==0.9.1
```
Comment (ssadhu-sl) [inline performance-test/requirements.txt:3]: Use requirements.in maybe from the start?

**Review by ssadhu-sl — APPROVED** [review]
looks good overall

File: performance-test/locustfile.py (hunk 1 above)
Comment (cpcundill) [reply performance-test/locustfile.py:35]: Yep - it's the default host which can be overridden when executing a test run.

File: performance-test/requirements.txt (hunk 2 above)
Comment (cpcundill) [reply performance-test/requirements.txt:3]: Good shout, I'll do that

**Review by GeorgeGoodall — APPROVED** [review]
lgtm

**Pull request merged**